pkg/registry/core/resourcequota/strategy.go

```diff
@@ -143,17 +144,33 @@ func (resourcequotaStatusStrategy) WarningsOnUpdate(ctx context.Context, obj, ol
 
 func getValidationOptionsFromResourceQuota(newObj *api.ResourceQuota, oldObj *api.ResourceQuota) validation.ResourceQuotaValidationOptions {
 	opts := validation.ResourceQuotaValidationOptions{
-		AllowPodAffinityNamespaceSelector: utilfeature.DefaultFeatureGate.Enabled(features.PodAffinityNamespaceSelector),
+		AllowPodAffinityNamespaceSelector:  utilfeature.DefaultFeatureGate.Enabled(features.PodAffinityNamespaceSelector),
+		AllowEphemeralStorageInScopedQuota: false,
 	}
 
 	if oldObj == nil {
 		return opts
 	}
 
 	opts.AllowPodAffinityNamespaceSelector = opts.AllowPodAffinityNamespaceSelector || hasCrossNamespacePodAffinityScope(&oldObj.Spec)
+	opts.AllowEphemeralStorageInScopedQuota = hasEphemeralStorage(&oldObj.Spec)
 	return opts
 }
 
+func hasEphemeralStorage(spec *api.ResourceQuotaSpec) bool {
```

Comment:
```suggestion
func hasScopedEphemeralStorage(spec *api.ResourceQuotaSpec) bool {
```

this also has to check if the old resource quota was scoped to core.ResourceQuotaScopeTerminating, core.ResourceQuotaScopeNotTerminating, core.ResourceQuotaScopeNotBestEffort, core.ResourceQuotaScopePriorityClass, core.ResourceQuotaScopeCrossNamespacePodAffinity, or core.ResourceQuotaScopeBestEffort

Reply:
and add a comment to set to true in 1.25

Reply:
1. According the document https://kubernetes.io/docs/concepts/policy/resource-quotas/#resource-quota-per-priorityclass，only the priorityclass scope is restricted to track ephemeralstorage，did we should check the other scope such as ResourceQuotaScopeNotTerminating？
2. should we check both Scopes and ScopeSelector？